odl/space/base_ntuples.py

```diff
@@ -64,23 +96,180 @@ def shape(self):
         """Shape ``(size,)`` of this space."""
         return (self.size,)
 
+    @property
+    def is_real(self):
+        """``True`` if the space represents R^n, i.e. real tuples."""
+        return self.__is_real and self.__is_floating
+
+    @property
+    def is_complex(self):
+        """``True`` if the space represents C^n, i.e. complex tuples."""
+        return (not self.__is_real) and self.__is_floating
+
+    @property
+    def is_numeric(self):
+        """``True`` if `dtype` is numeric, otherwise ``False``."""
+        return self.__is_real or self.__is_complex
+
+    @property
+    def real_dtype(self):
+        """The real dtype corresponding to this space's `dtype`.
+
+        Raises
+        ------
+        NotImplementedError
+            If `dtype` is not a numeric data type.
+        """
+        if not self.is_numeric:
+            raise NotImplementedError(
+                '`real_dtype` not defined for non-numeric `dtype`')
+        return self.__real_dtype
+
+    @property
+    def complex_dtype(self):
+        """The complex dtype corresponding to this space's `dtype`.
+
+        Raises
+        ------
+        NotImplementedError
+            If `dtype` is not a numeric data type.
+        """
+        if not self.is_numeric:
+            raise NotImplementedError(
+                '`complex_dtype` not defined for non-numeric `dtype`')
+        return self.__complex_dtype
+
+    @property
+    def real_space(self):
```

Comment:
Hm, you're right. Why invent new names?

Reply:
Oh, wait. `real` is, of course, okay, but what about `complex`? There is no such thing for vectors. I'll stick to these names since it's not the same.

Reply:
Well wouldn't `real`/`complex` still be an improvement?